src/state.rs

```diff
@@ -8,53 +8,60 @@ use opentelemetry::metrics::{Counter, UpDownCounter};
 use opentelemetry::sdk::trace::Tracer;
 use std::sync::Arc;
 use tracing_subscriber::prelude::*;
+use crate::stores::tenant::TenantStore;
 
 #[derive(Clone)]
 pub struct Metrics {
     pub registered_webhooks: UpDownCounter<i64>,
     pub received_notifications: Counter<u64>,
 }
 
-pub trait State<C, N>
-where
-    C: ClientStore,
-    N: NotificationStore,
+pub trait State<C, N, T>
+    where
+        C: ClientStore,
+        N: NotificationStore,
+        T: TenantStore,
 {
     fn config(&self) -> Config;
     fn build_info(&self) -> BuildInfo;
     fn client_store(&self) -> C;
     fn notification_store(&self) -> N;
+    fn tenant_store(&self) -> T;
     fn providers(&self) -> Providers;
     fn supported_providers(&self) -> Vec<ProviderKind>;
     fn relay_client(&self) -> RelayClient;
 }
 
 #[derive(Clone)]
-pub struct AppState<C, N>
-where
-    C: ClientStore,
-    N: NotificationStore,
+pub struct AppState<C, N, T>
```

Comment:
We should refactor this to remove type parameters, and instead use `client_store: Arc<dyn ClientStore>` etc.

Reply:
Refactored!